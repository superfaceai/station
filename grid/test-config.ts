import {
  NockConfig,
  SuperfaceTest,
  SuperfaceTestConfig,
} from '@superfaceai/testing';

export const nockConfig: NockConfig = {
  testInstance: expect,
};

/* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
export const buildSuperfaceTest = (global as any).buildSuperfaceTest as (
  options?: SuperfaceTestConfig
) => SuperfaceTest;
