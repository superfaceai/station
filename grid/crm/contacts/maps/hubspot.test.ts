import { SuperfaceTest } from '@superfaceai/testing';
import { nockConfig } from '../../../test-config';

describe(`crm/contacts/hubspot`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest(
      {
        profile: 'crm/contacts',
        provider: 'hubspot',
      },
      nockConfig
    );
  });

  describe('Create', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'Create',
          input: {
            email: 'test@example.com',
            firstName: 'Test',
            lastName: 'User',
            company: 'Station Test',
            country: 'USA',
            customProperties: {
              myproperty: 'value',
            },
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });

  describe('Update', () => {
    it('should return error if id is missing', async () => {
      await expect(
        superface.run({
          useCase: 'Update',
          input: {
            email: 'test@example.com',
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'Update',
          input: {
            id: '401',
            firstName: 'Updated first name',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });

  describe('Search', () => {
    it('should return error for unknown operator', async () => {
      await expect(
        superface.run({
          useCase: 'Search',
          input: {
            property: 'firstname',
            operator: '%',
            value: 'John',
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'Search',
          input: {
            property: 'firstname',
            operator: 'EQ',
            value: 'Test',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
