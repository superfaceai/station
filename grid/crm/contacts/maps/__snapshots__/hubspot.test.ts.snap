// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crm/contacts/hubspot Create should perform successfully 1`] = `
Ok {
  "value": Object {
    "id": "651",
  },
}
`;

exports[`crm/contacts/hubspot Search should perform successfully 1`] = `
Ok {
  "value": Array [],
}
`;

exports[`crm/contacts/hubspot Search should return error for unknown operator 1`] = `
Err {
  "error": "InputValidationError: Input validation failed:
Path: input.operator
Error: Invalid enum value: \\"%\\"",
}
`;

exports[`crm/contacts/hubspot Update should perform successfully 1`] = `
Ok {
  "value": Object {
    "id": "401",
  },
}
`;

exports[`crm/contacts/hubspot Update should return error if id is missing 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Missing field\\",
  \\"detail\\": \\"ID is missing in input values\\"
}
AST Path: definitions[1].statements[0]
Original Map Location: Line 52, column 3",
}
`;
